core/templates/dev/head/app.js

```diff
@@ -40,17 +40,18 @@ oppia.constant(
 oppia.constant('PLACEHOLDER_OUTCOME_DEST', '/');
 oppia.constant('INTERACTION_DISPLAY_MODE_INLINE', 'inline');
 oppia.constant('DEFAULT_RULE_NAME', 'Default');
-oppia.constant('CLASSIFIER_RULESPEC_STR', 'FuzzyMatches');
+oppia.constant('RULE_TYPE_CLASSIFIER', 'FuzzyMatches');
 oppia.constant('OBJECT_EDITOR_URL_PREFIX', '/object_editor_template/');
 // Feature still in development.
 // NOTE TO DEVELOPERS: This should be synchronized with the value in feconf.
 oppia.constant('ENABLE_STRING_CLASSIFIER', false);
-oppia.constant('DEFAULT_CLASSIFIER_RULE_SPEC', {
+oppia.constant('DEFAULT_CLASSIFIER_RULE', {
```

Comment:
So I don't think this constant is needed anywhere besides the RuleObjectFactory -- let's make it internal to the function that creates a new classifier rule; the fewer global constants we need to keep track of, the better. Searching for it, there's an unused import as well, and a test in BaseSpec, but you can replace the latter with a test for RuleObjectFactory that creates a new classifier rule and validates its type in the same way.